src/ebonite/core/objects/core.py

```diff
@@ -652,3 +677,36 @@ def environment(self, environment: RuntimeEnvironment):
         if not isinstance(environment, RuntimeEnvironment):
             raise ValueError(f'{environment} is not RuntimeEnvironment')
         self.environment_id = environment.id
+
+    def bind_runner(self, runner):
+        self.runner = runner
+        return self
+
+    def unbind_runner(self):
+        del self.runner
+
+    @property
+    def has_runner(self):
+        return self.runner is not None
+
+    @_with_runner
+    def logs(self, **kwargs):
+        """
+
+        :param kwargs: parameters for runner `logs` method
+        :yields: str logs from running instance
+        """
+        yield from self.runner.logs(self.params, self.environment.params, **kwargs)
+
+    @_with_runner
+    def is_running(self, **kwargs) -> bool:
+        """
+        Checks whether instance is running
+
+        :param kwargs: params for runner `is_running` method
+        :return: "is running" flag
+        """
+        if not self.has_meta_repo:
+            return False # TODO separate repo logic from runner logic and remove this check
```

Comment:
In this PR persistence primitives (which are part of repo logic) were bound to runners. For me it looks like this is the main purpose of the PR. So then what does this TODO mean in this context? Reverting this PR changes?

Reply:
No :)
For now, actual instance state is bound to it's repo state - it can only be persisted if service is running. if you stop it, you must delete it. 
Yes, it is very simple, but also very limited. For example, what if service failed to start? Or crashed? There is no way to synchronize it status with repository. I think we need to separate this states, maybe even add fields to instace model like "running", "error_msg" and so on - I'll add a task for it

Reply:
In fact no.

Take a look at `Ebonite.run_instance` and `Ebonite.stop_instance` methods. I implemented them as they are on purpose. In `run` we save instance to repo **before** actual running to ensure that it's state will be tracked no matter what happened at initialization or runtime. In `stop` we delete instance from repo **after** we are sure that it is stopped correctly to ensure no ghost containers in the system.

This way repo has really no clue about status of instance. And I think it shouldn't synchronize anything as there already is a source of truth for this: runner. If we try to synchronize we will inevitably face consistency ("stale" status) problems.

Reply:
I hope you get my point and propose to move discussion (if needed) to chats. This PR could be approved.